src/main/java/audio/harmonizer/Harmonizer.java

```diff
@@ -207,14 +229,36 @@ private int[] normalizeDelayReverb() throws AppError {
         int MAX_DELAY_REVERB = Constants.SAMPLE_RATE / 20; // number of samples corresponding to 50ms
         int DEFAULT_DELAY_REVERB = Constants.SAMPLE_RATE / 25; // number of samples corresponding to 40ms
         double[] delayReverb = dataRaw.getDelayReverb();
-        if (delayReverb != null) {
-            int[] output = new int[delayReverb.length];
-            for (int i = 0; i < delayReverb.length; i++) {
-                checkDouble(delayReverb[i], "delayReverb", i);
+        int maxNoDelayValues = (int) (1.0/3 * (numberBeats / (Constants.TEMPO / 60f)));
 
-                output[i] = delayReverb[i] == -1 ? DEFAULT_DELAY_REVERB : (int) (delayReverb[i] * MAX_DELAY_REVERB);
+        if (delayReverb != null) {
```

Comment:
This looks like a lot of code that's very similar to the echo code, can we factor it out somehow?